packages/js/product-editor/src/hooks/use-product-template/use-product-template.ts

```diff
@@ -59,23 +59,21 @@ function findBetterMatchTemplate( matchingTemplates: ProductTemplate[] ) {
 
 export const useProductTemplate = (
 	productTemplateId: string | undefined,
-	product: Partial< Product > | undefined | null
+	product: Partial< Product > | null
 ) => {
 	const productTemplates =
 		window.productBlockEditorSettings?.productTemplates ?? [];
 
-	const productType = product?.type;
+	// we shouldn't default to the standard-product-template for variations
+	if ( ! productTemplateId && product?.type === 'variation' ) {
+		return { productTemplate: null, isResolving: false };
```

Comment:
Hmm this seemed to work anyhow, is there logic somewhere else that defaults to the `variation` template?

Reply:
Aaah I see we have a different fallback somewhere else: https://github.com/woocommerce/woocommerce/blob/c1900065f2aafdde1fd6d72c46e0df435ba13542/packages/js/product-editor/src/components/block-editor/block-editor.tsx#L75

Could we potentially remove the other exceptions, so there is a single place where we fall back to `product-variation` as the template if it is a variation? Like in the logic above for example?
Having multiple places where we specifically check for variations seems error prone and also confusing.

Reply:
What started happening is that I added a product template with the type 'variation' for gift card variations. Regular variations were being matched to the "gift card variation" product template, since we don't have a product template for regular variations. Previously, the `useProductTemplate` hook returned undefined. 

I agree it's kind of confusing, but I prefer leaving it like this for now, because we have layout templates and product templates.